modules/core/src/main/scala/sangria/execution/FutureResolver.scala

```diff
@@ -1572,8 +1693,7 @@ private[execution] class FutureResolver[Ctx](
               deferredResolverState = deferredResolverState
             )
 
-            if (allFields.exists(_.deprecationReason.isDefined))
-              deprecationTracker.deprecatedFieldUsed(ctx)
+            deprecationTracker.map(trackDeprecation(_, ctx))
```

Comment:
We're returning `Unit`, so I think no need to map it.
```suggestion
            deprecationTracker.foreach(trackDeprecation(_, ctx))
```

Reply:
habit from mostly working with IO stuff 😅 updating